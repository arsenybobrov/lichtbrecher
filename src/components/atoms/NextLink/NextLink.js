import React from 'react';
import PropTypes from 'prop-types';
import Link from 'next/link';
import linkResolver from '../../../../api/prismic/linkResolver';

const NextLink = ({
  text, url, templatePath, title = text, target,
}) => (
  <Link as={linkResolver(url)} href={templatePath}>
    <a
      href={linkResolver(url)}
      title={title}
      target={target}
      rel={target === '_blank' ? 'noopener norefferer' : null}
    >
      {text}
    </a>
  </Link>
);

NextLink.propTypes = {
  text: PropTypes.string,
  url: PropTypes.object,
  templatePath: PropTypes.string,
  title: PropTypes.string,
  target: PropTypes.oneOf(['_blank', '_self']),
};

export default NextLink;
