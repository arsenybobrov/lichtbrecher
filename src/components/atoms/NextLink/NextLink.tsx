import React from 'react';
import Link from 'next/link';
import get from 'lodash/get';
import linkResolver from '../../../../api/prismic/helper/linkResolver';

interface Props {
  text: string,
  url: {
    type: string;
  },
  title: string,
}

const NextLink: React.FC<Props> = ({
  text, url, title = text,
}) => {
  let href = '';
  if (get(url, 'type', null)) {
    href = url.type === 'homepage' ? '/' : '/[page]';
  }

  return (
    <Link
      as={linkResolver(url)}
      href={href}
    >
      <a
        href={linkResolver(url)}
        title={title}
        target={get(url, 'target', null)}
        rel={get(url, 'target', '_self') === '_blank' ? 'noopener noreferrer' : undefined}
      >
        {text}
      </a>
    </Link>
  );
};

export default NextLink;
