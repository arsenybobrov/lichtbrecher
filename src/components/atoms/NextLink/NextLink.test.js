import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import NextLink from './NextLink';
import {
  prismicExternalLinkMock,
  prismicInternalHomepageLinkMock,
  prismicInternalPageLinkMock,
} from '../../../../api/prismic/tests/linkResolver.test';

export const NextLinkInternalPageLinkMock = {
  url: prismicInternalPageLinkMock,
  title: 'here comes the title',
  text: 'Click me',
};

export const NextLinkInternalHomepageLinkMock = {
  url: prismicInternalHomepageLinkMock,
  title: 'here comes the title',
  text: 'Click me',
};

export const NextLinkExternalLinkMock = {
  url: prismicExternalLinkMock,
  text: 'Click me',
};

describe('NextLink snapshot', () => {
  it.each`
    name                                             | mock
    ${'internal Link to a page'}                     | ${NextLinkInternalPageLinkMock}
    ${'internal Link to the homepage'}               | ${NextLinkInternalHomepageLinkMock}
    ${'external link with target _blank'}            | ${NextLinkExternalLinkMock}
  `('$name', ({ mock }) => {
  const tree = renderer
    .create(
      <NextLink {...mock} />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
});
