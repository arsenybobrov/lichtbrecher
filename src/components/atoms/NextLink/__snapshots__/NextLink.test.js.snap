// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NextLink snapshot external link with target _blank 1`] = `
<a
  href="https://example.com"
  onClick={[Function]}
  onMouseEnter={[Function]}
  rel="noopener noreferrer"
  target="_blank"
  title="Click me"
>
  Click me
</a>
`;

exports[`NextLink snapshot internal Link to a page 1`] = `
<a
  href="/page-two"
  onClick={[Function]}
  onMouseEnter={[Function]}
  rel={null}
  target={null}
  title="here comes the title"
>
  Click me
</a>
`;

exports[`NextLink snapshot internal Link to the homepage 1`] = `
<a
  href="/"
  onClick={[Function]}
  onMouseEnter={[Function]}
  rel={null}
  target={null}
  title="here comes the title"
>
  Click me
</a>
`;
